File: data/src/main/java/com/pokemon/data/sharedPreferences/PreferencesManager.java
package com.pokemon.data.sharedPreferences;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Inject;
import javax.inject.Singleton;

import static android.content.Context.MODE_PRIVATE;

@Singleton
public class PreferencesManager {

    private static final String KEY_ = "key_";
    private final Context context;


    @Inject
    public PreferencesManager(Context context) {
        this.context = context;
    }

    public void save(String input) {
        SharedPreferences.Editor editor = context.getSharedPreferences(KEY_, MODE_PRIVATE).edit();
        editor.putString(KEY_, input);
        editor.apply();
    }

    public int getLastAppointmentId() {
        SharedPreferences prefs = context.getSharedPreferences(KEY_, MODE_PRIVATE);
        return prefs.getInt(KEY_, 0);
    }

}

File: data/src/main/java/com/pokemon/data/sharedPreferences/PreferencesManager.kt
package com.pokemon.data.sharedPreferences

import android.content.Context
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class PreferencesManager @Inject constructor(private val context: Context) {
    fun save(input: String?) {
        val editor = context.getSharedPreferences(
            PREFERENCES,
            Context.MODE_PRIVATE
        ).edit()
        editor.putString(KEY_USERNAME, input)
        editor.apply()
    }

    fun clear() {
        val editor = context.getSharedPreferences(
            PREFERENCES,
            Context.MODE_PRIVATE
        ).edit()
        editor.clear()
        editor.apply()
    }

    val getUsername: String?
        get() {
            val prefs = context.getSharedPreferences(
                PREFERENCES,
                Context.MODE_PRIVATE
            )
            return prefs.getString(KEY_USERNAME, "")
        }

    companion object {
        private const val PREFERENCES = "pokemon"
        private const val KEY_USERNAME = "key_username"
    }
}